### PR TITLE
moving all const to config package

### LINKED ISSUE #99: move all config related code to `config` package
add into this PR- https://github.com/Real-Dev-Squad/tiny-site-backend/pull/98/files

ref- https://github.com/Real-Dev-Squad/wisee-backend/blob/develop/src/utils/jwt.go#L12

### PR DESCRIPTION
**Pull request by iamitprakash**
## Issue Ticket Number

-   closes #99 

## Description

Moving all const to config package and use config instead of os.getenv

**Breaking Changes**

-   [ ] Yes
-   [x] No

_If your feature introduces breaking changes or if something is missing, please mention the related issue tickets._

**Development Tested?**

-   [x] Yes
-   [ ] No

_Confirm whether the changes have been tested locally during development._

**Tested in Staging?**

-   [ ] Yes
-   [x] No

_Indicate whether the changes have been tested in the staging environment for dev to main._

**Database Changes**

-   [ ] Yes
-   [x] No

_Indicate whether the changes include modifications to the database._

### Screenshots

Attach any relevant screenshots, such as test coverage reports, before and after images, or other visual aids.

## Additional Notes

Include any additional notes, considerations, or explanations that might be helpful for reviewers.

### CODE REVIEW

File: controllers/auth.go [hunk 1]
```diff
@@ -125,8 +123,8 @@ func GoogleCallback(ctx *gin.Context, db *bun.DB) {
         ctx.JSON(http.StatusInternalServerError, gin.H{"message": "error"})
         return
     }
-
-    ctx.SetCookie("token", token, tokenExpiration, "/", domain, true, true)
+    fmt.Println("config token expiration", config.TokenExpiration)
```
Comment (iamitprakash) [inline controllers/auth.go:126]: do we need to print this

File: controllers/url.go [hunk 2]
```diff
@@ -58,24 +66,38 @@ func CreateTinyURL(ctx *gin.Context, db *bun.DB) {
...
-	intVar, err := strconv.ParseInt(config.Max_Url_Count, 0, 64)
+	if err != nil {
+		ctx.JSON(http.StatusInternalServerError, dtos.URLCreationResponse{
+			Message: "Failed to check URL count",
+		})
+		return
+	}
 
-	if err != nil && int64(count) >= intVar {
+	fmt.Println("count ++++", count)
```
Comment (iamitprakash) [inline controllers/url.go:91]: same here

File: controllers/url.go [hunk 3]
```diff
@@ -58,24 +66,38 @@ func CreateTinyURL(ctx *gin.Context, db *bun.DB) {
...
-	body.CreatedAt = time.Now().UTC()
+	// Count the number of existing URLs for the user
+	count, err := db.NewSelect().
+		Model(&models.Tinyurl{}).
+		Where("user_id = ?", body.UserID).
+		Where("is_deleted = ?", false).
+		Count(ctx)
 
-	intVar, err := strconv.ParseInt(config.Max_Url_Count, 0, 64)
+	if err != nil {
```
Comment (iamitprakash) [inline controllers/url.go:84]: why we can't merge #84  and #94, having mentioned that do we really need to show error mentioned in line 86 to user

File: controllers/auth.go (hunk 1 above)
Comment (vinit717) [reply controllers/auth.go:126]: No doing for testing the app  will remove after fixing the failing test

File: controllers/url.go (hunk 3 above)
Comment (vinit717) [reply controllers/url.go:84]: Ok I'll merge them

**Review by vinit717 — APPROVED** [review]

**Pull request merged**